docs/src/components/mdx/code-block.tsx

```diff
@@ -0,0 +1,127 @@
+import Button from '@pluralsight/ps-design-system-button'
+import Theme, { useTheme } from '@pluralsight/ps-design-system-theme'
+import cx from 'classnames'
+import Prism from 'prismjs/components/prism-core'
+import Highlight, { PrismTheme, defaultProps } from 'prism-react-renderer'
+import React, { HTMLAttributes, useEffect, useState } from 'react'
+import { CopyToClipboard } from 'react-copy-to-clipboard'
+
+import styles from './code-block.module.css'
+import { darkTheme, lightTheme } from './code-block-theme'
+
+interface CodeBlockProps extends HTMLAttributes<HTMLDivElement> {
+  live?: boolean
+}
+
+export const CodeBlock: React.FC<CodeBlockProps> = (props) => {
```

Comment:
initially i was using the remark-prism plugin but it was getting in my way so i'm now using this custom component with prism-react-rendered. this will allow us a greater level of flexibility